doc-bridgepoint/notes/9706_polys/9706_polys_dnt.md

```diff
@@ -0,0 +1,244 @@
+polymorphic event used in the supertype).
+
+5.3 All events are polymorphic events
+
+In the "concrete polymorphic" idiom, all events are polymorphic -- that is to
+say, an event _must_ be handled in every instance state machine in a
+subtype/supertype hierarchy. Because of this, a column in the state event matrix
+must be created immediately if a supertype event is created. The column will
+contain initial values for the cells ("Can't Happen" or "Ignore"), but these can
+be changed or assigned to a transition. The current BridgePoint behavior is to
+only create the column of the state event matrix when a polymorphic event is
+actually assigned to a transition. This new behavior must be invoked to sync the
+columns at the following actions:
+
+5.3.1 Create event  
+5.3.2 Paste event  
+5.3.3 Create state machine in subtype  
+5.3.4 Add subtype to subtype/supertype hierarchy  
+5.3.5 Unassign event from creation transition (see section 5.5)  
+
+5.3.6 Can non-local events be deleted?
+
+Currently BridgePoint allows the deletion of non-local events. In the abstract
+idiom, this makes sense as non-local events only ever apear if it is used in a
+transition. Deleting such an event results in all of the transitions it
+participates in being unassigned.
+
+In the concrete case, this does not make sense since an event _must_ be handled
+in every subtype state machine (include "Cannot Happen" or "Ignore"). Because
+of this property, deletion of non-local events shall be disallowed when
+concrete polymorphic events are enabled.
+
+5.4 UI Masking in the supertype
+
+This strategy of using a non-local event referring to a local polymorphic event
+in the supertype results in two "identical" events in the supertype. Some
+special masking will need to be done to prevent both from showing in the tree.
+Additionally, the naming convention for local events should be followed (as
+opposed to scoped naming as for polymorphic events). Special cases shall be put
+into place to make this look correct. Additionally, the state event matrix must
+not have a column for each of the two events, but just one. Integrity errors
+that warn two identical events in the same class will be suppressed when
+concrete polymorphic events are used.
+
+5.5 Creation events
+
+As mentioned in 5.3, "all events are polymorphic". This means that creation
+transitions will have no valid events to be assigned because every newly created
+event will be converted to a polymorphic event immediately. The UI will be
+relaxed to allow assigning polymorphic events to creation transitions, however,
+the action will be cancelled if any of the non-local events in the subtypes are
+assigned to a transition. Once selected for assignment, the polymorphic event
+will be migrated to a local event.
+
+Additionally, when an event is unassigned from a creation transition, the sync
+will be invoked to re-migrated to a polymorphic event and propagated to the
```

Comment:
grammar error in first sentence around "invoked to re-migrated to"

Reply:
done